lib/client.js

```diff
@@ -1,55 +1,131 @@
-'use strict';
+const bluebird = require('bluebird');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const redis = require('redis');
+const Queue = require('./queue.js');
+const Job = require('./job.js');
+const logger = require('./logger.js');
 
-/**
- * Basic qless client object for interacting with qless.
- */
+bluebird.promisifyAll(fs);
+bluebird.promisifyAll(redis.RedisClient.prototype);
 
-const redis = require('redis');
-const os = require('os');
-const process = require('process');
-
-const luaScript = require('./lua_script');
-const queue = require('./queue');
-const jobs = require('./jobs');
-
-class Client {
-  /**
-   * Parameters are passed through to redis.createClient()
-   */
-  constructor() {
-    this.redis = redis.createClient.apply(redis, arguments);
-    this._lua = new luaScript.LuaScript('qless', this.redis);
-    this.jobs = new jobs.Jobs(this);
-    this.workerName = [os.hostname(), process.env.HOST, process.pid.toString()].join('-');
-  }
+function Client(config) {
+  this.config = config;
+  this.redis = redis.createClient(config);
+  this.path = path.join(__dirname, 'qless-core', 'qless.lua');
+  this.sha = null;
+  this.workerName = config.hostname || os.hostname();
+}
 
-  /**
-   * Args: command, arg1, arg1, ..., cb
-   */
-  call() {
-    // convert arguments to real array
-    const args = Array.prototype.slice.call(arguments, 0);
-    const cb = args.pop();
-
-    // Add "time" between command and rest of args
-    const time = new Date().getTime() / 1000.0;
-    args.splice(1, 0, time);
-    this._lua.call(args, cb);
-  }
+// Return a new disposer
+Client.disposer = function(config) {
+  return new bluebird.Promise((resolve, reject) => {
```

Comment:
Ended up with `return Promise.resolve(new Client(config)).disposer(client => client.quit());`

And yes, the linting _did_ complain once I got the invocation right.